docs/design/2021-04-18-common-table-expression.md

```diff
@@ -0,0 +1,276 @@
+<!--
+This is a template for TiDB's change proposal process, documented [here](./README.md).
+-->
+
+# Proposal: Support Common Table Expression
+
+- Author(s):     [@guo-shaoge](https://github.com/guo-shaoge) [@wjhuang2016](https://github.com/wjhuang2016) 
+- Last updated:  2021-04-18
+- Discussion at: https://github.com/pingcap/tidb/issues/17472 <!-- https://github.com/pingcap/tidb/issues/XXX -->
```

Comment:
```suggestion
- Author(s): [@guo-shaoge](https://github.com/guo-shaoge), [@wjhuang2016](https://github.com/wjhuang2016) 
- Discussion PR: https://github.com/pingcap/tidb/pull/24147
- Tracking Issue: https://github.com/pingcap/tidb/issues/17472
```

Reply:
And I think we don't have a real tracking issue because #17472 doesn't track effort. We don't have to use a project but subtasks would better to refer the tracking issue #17472 .

Reply:
done